src/core/interpreter.cc

```diff
@@ -895,4 +895,12 @@ void InterpreterManager::Return(Interpreter* ir) {
   waker_.notify();
 }
 
+void InterpreterManager::Reset() {
+  waker_.await([this]() { return available_.size() == storage_.size(); });
+
+  for (auto& interpreter : storage_) {
```

Comment:
actually I see that we need to reset the storage_ and the available_ as available holds pointers to interperters that are in storage_
check out the InterpreterManager::Get() logic
so you dont need to go over storage_ and call constructor again. reset will be enough